### PR TITLE
{bio}[GCC/12.2.0] Bismark v0.24.0, Bowtie2 v2.5.1 w/ Python 3.10.8

### PR DESCRIPTION
**Pull request by eylenth**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/8dbcb04cfa444527864966d59b619fec for a full test report.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/b/Bismark/Bismark-0.24.0-GCC-12.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Authors:: Thekla Loizou <t.loizou@cyi.ac.cy>
+# License:: MIT/GPL
+#
+# Updated to 0.23.1
+# J. Sassmannshausen NHS/GSTT
+
+easyblock = 'Tarball'
+
+name = 'Bismark'
+version = '0.24.0'
```
Comment (smoors) [inline easybuild/easyconfigs/b/Bismark/Bismark-0.24.0-GCC-12.2.0.eb:14]: version `0.24.1` has been released 3 weeks ago, can we use that?

Comment (eylenth) [reply easybuild/easyconfigs/b/Bismark/Bismark-0.24.0-GCC-12.2.0.eb]: version is now upgraded to `0.24.1`

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Comment by smoors** [comment]
Going in, thanks @eylenth!

**Comment by garadar** [comment]
During compilation of Bowtie2, I get an error message for missing libnsl:

```
== Temporary log file in case of crash /opt/cluster/easybuild/logs/easybuild-e3hgby73.log
== found valid index for /opt/ebsofts/EasyBuild/4.8.0/easybuild/easyconfigs, so using it...
== resolving dependencies ...
== processing EasyBuild easyconfig /home/xxx/easybuild/Bowtie2-2.5.1-GCC-12.2.0.eb
== building and installing Compiler/GCC/12.2.0/Bowtie2/2.5.1...
  >> installation prefix: /opt/ebsofts/Bowtie2/2.5.1-GCC-12.2.0
== fetching files...
  >> sources:
  >> /opt/cluster/easybuild/ebfiles_repo/b/Bowtie2/bowtie2-2.5.1-source.zip [SHA256: cb6cbbbb5a7167a2f21a3d63cb9774336361f540e1ec3d8ff907f955c35f71b8]
== ... (took < 1 sec)
== creating build dir, resetting environment...
  >> build dir: /dev/shm/ebbuild/Bowtie2/2.5.1/GCC-12.2.0
== ... (took < 1 sec)
== unpacking...
  >> running command:
	[started at: 2023-07-12 10:02:43]
	[working dir: /dev/shm/ebbuild/Bowtie2/2.5.1/GCC-12.2.0]
	[output logged in /tmp/eb-pi1lz4l7/easybuild-run_cmd-8g9d10f2.log]
	unzip -qq /opt/cluster/easybuild/ebfiles_repo/b/Bowtie2/bowtie2-2.5.1-source.zip
  >> command completed: exit 0, ran in < 1s
== ... (took < 1 sec)
== patching...
== ... (took < 1 sec)
== preparing...
  >> loading toolchain module: GCC/12.2.0
  >> (no build dependencies specified)
  >> loading modules for (runtime) dependencies:
  >>  * zlib/1.2.12
  >>  * Perl/5.36.0
  >>  * Python/3.10.8
  >> defining build environment for GCC/12.2.0 toolchain
== ... (took 9 secs)
== configuring...
== ... (took < 1 sec)
== building...
  >> running command:
	[started at: 2023-07-12 10:02:53]
	[working dir: /dev/shm/ebbuild/Bowtie2/2.5.1/GCC-12.2.0/bowtie2-2.5.1]
	[output logged in /tmp/eb-pi1lz4l7/easybuild-run_cmd-za5xnx3e.log]
	make  -j 12  CC="gcc"  CPP="g++" CXX="g++"  RELEASE_FLAGS="-O2 -ftree-vectorize -march=core2 -fno-math-errno -fPIC -std=gnu++98"
  >> command completed: exit 0, ran in 00h00m52s
== ... (took 52 secs)
== testing...
== ... (took < 1 sec)
== installing...
== ... (took < 1 sec)
== taking care of extensions...
== ... (took < 1 sec)
== restore after iterating...
== ... (took < 1 sec)
== postprocessing...
== ... (took < 1 sec)
== sanity checking...
  >> file 'bin/bowtie2' found: OK
  >> file 'bin/bowtie2-build' found: OK
  >> file 'bin/bowtie2-inspect' found: OK
  >> file 'bin/MANUAL' found: OK
  >> file 'bin/MANUAL.markdown' found: OK
  >> file 'bin/NEWS' found: OK
  >> file 'bin/bowtie2-align-l' found: OK
  >> file 'bin/bowtie2-align-s' found: OK
  >> file 'bin/bowtie2-build-l' found: OK
  >> file 'bin/bowtie2-build-s' found: OK
  >> file 'bin/bowtie2-inspect-l' found: OK
  >> file 'bin/bowtie2-inspect-s' found: OK
  >> file 'bin/LICENSE' found: OK
  >> (non-empty) directory 'doc' found: OK
  >> (non-empty) directory 'example' found: OK
  >> (non-empty) directory 'scripts' found: OK
  >> loading modules: GCC/12.2.0, Bowtie2/2.5.1...
  >> running command 'bowtie2 --help' ...
  >> result for command 'bowtie2 --help': FAILED
  >> running command 'bowtie2-build --help' ...
  >> result for command 'bowtie2-build --help': OK
== ... (took 12 secs)
== FAILED: Installation ended unsuccessfully (build directory: /dev/shm/ebbuild/Bowtie2/2.5.1/GCC-12.2.0): build failed (first 300 chars): Sanity check failed: sanity check command bowtie2 --help exited with code 127 (output: perl: error while loading shared libraries: libnsl.so.1: cannot open shared object file: No such file or directory
) (took 1 min 15 secs)
== Results of the build can be found in the log file(s) /opt/cluster/easybuild/logs/easybuild-Bowtie2-2.5.1-20230712.100243.YrikZ.log
ERROR: Build of /home/xxx/easybuild/Bowtie2-2.5.1-GCC-12.2.0.eb failed (err: 'build failed (first 300 chars): Sanity check failed: sanity check command bowtie2 --help exited with code 127 (output: perl: error while loading shared libraries: libnsl.so.1: cannot open shared object file: No such file or directory\n)')

```

I had to libnsl in dependencie:
```
dependencies = [
    ('zlib', '1.2.12'),
    ('Perl', '5.36.0'),
    ('Python', '3.10.8'),
    ('libnsl', '2.0.0'),
]
```

So I created a new PR #18293 

Could you tell me if I did something wrong ?

**Comment by verdurin** [comment]
@garadar which OS are you building on?

**Comment by garadar** [comment]
I use apptainer with rocky8:

```
apptainer exec \
   --nv \
   --bind $EBFILES_REPO \
   --bind $EB_DIR \
   --bind $EBSOFTS_SRC:$EBSOFTS_DST:rw \
   --bind $EBMODULES_SRC:$EBMODULES_DST:rw \
   docker://ghcr.io/easybuilders/rockylinux-8.7-amd64 \
   bash -c "ml EasyBuild; eb ${EB_OPTS}"
```

**Comment by branfosj** [comment]
```
perl: error while loading shared libraries: libnsl.so.1: cannot open shared object file: No such file or directory
```
The error is on running Perl, so I expect you should be seeing the issue whenever Perl is used. However, I'm not sure why there is link from Perl to libnsl. I am not seeing it
```
$ ldd perl
        linux-vdso.so.1 (0x00007ffda896f000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fe2e863b000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fe2e8437000)
        libm.so.6 => /lib64/libm.so.6 (0x00007fe2e80b5000)
        libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007fe2e7e8c000)
        libutil.so.1 => /lib64/libutil.so.1 (0x00007fe2e7c88000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fe2e78c3000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fe2e885b000)
```

**Pull request merged**